### PR TITLE
Added the algorithm of hamming distance in others inside newly created cn package

### PR DESCRIPTION
**Pull request by MihSawant**
Hamming distance an algorithm used in **Computer Networking** which finds total no.of error bits between the sender data bits and the receiver data bits.

### CODE REVIEW

**Comment by MihSawant** [comment]
@siriak, can you please review the code uploaded

File: pom.xml [hunk 1]
```diff
@@ -61,8 +61,8 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
                 <configuration>
-                    <source>17</source>
-                    <target>17</target>
+                    <source>16</source>
+                    <target>16</target>
```
Comment (siriak) [inline pom.xml:65]: Please revert this change

File: src/main/java/com/thealgorithms/others/cn/HammingDistance.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        byteArray.add(senderBits.getBytes());
+        byteArray.add(receiverBits.getBytes());
+
+        return byteArray;
+    }
+
+    public int getHammingDistanceBetweenBits(String senderBits, String receiverBits) {
+
+        if(senderBits.length() != receiverBits.length()) {
+           return -1;
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/cn/HammingDistance.java:21]: Please raise an exception instead

File: src/main/java/com/thealgorithms/others/cn/HammingDistance.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+        for(int i = 0; i < senderData.length; i++){
+           totalErrorBitCount += senderData[i] ^ receiverData[i];
+        }
+
+        if(totalErrorBitCount == 0){
+            System.out.println("No Error bit in data segments");
+        } else{
+            System.out.println("Total Error bit count "+totalErrorBitCount);
+        }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/cn/HammingDistance.java:38]: This logging is redundant

File: src/main/java/com/thealgorithms/others/cn/HammingDistance.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+    private List<byte[]> extractBinaryData(String senderBits, String receiverBits){
+        List<byte[]> byteArray = new ArrayList<>();
+
+        byteArray.add(senderBits.getBytes());
+        byteArray.add(receiverBits.getBytes());
+
+        return byteArray;
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/others/cn/HammingDistance.java:16]: Please remove this redundant method

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/others/cn/HammingDistance.java (hunk 2 above)
Comment (MihSawant) [reply src/main/java/com/thealgorithms/others/cn/HammingDistance.java]: Yes, reverted the change

File: src/main/java/com/thealgorithms/others/cn/HammingDistance.java (hunk 4 above)
Comment (MihSawant) [reply src/main/java/com/thealgorithms/others/cn/HammingDistance.java]: Removed the redundant method and moved array inside getHammingDistance() method.

**Review by siriak — APPROVED** [review]
